lisp/ess-roxy.el

```diff
@@ -607,15 +606,17 @@ in a temporary buffer and return that buffer."
                     (not (looking-at ess-roxy-re))))
         (append-to-file beg (point) tmpf))
       (ess-force-buffer-current)
-      (ess-command (concat "print(suppressWarnings(require(" ess-roxy-package
-                           ", quietly=TRUE)))\n")
-                   roxy-buf)
+      (unless (ess-boolean-command (concat "print(suppressWarnings(require(" ess-roxy-package
+                                           ", quietly=TRUE)))\n"))
+        (error (concat "Failed to load the " ess-roxy-package " package; "
+                       "in R, try  install.packages(\"" ess-roxy-package "\")")))
+      (ess-command (concat out-rd-roclet "(\"" tmpf "\")\n") roxy-buf)
       (with-current-buffer roxy-buf
-        (goto-char 1)
-        (if (search-forward-regexp "FALSE" nil t)
-            (error (concat "Failed to load the " ess-roxy-package " package; "
-                           "in R, try  install.packages(\"" ess-roxy-package "\")"))))
-      (ess-command (concat out-rd-roclet "(\"" tmpf "\")\n") roxy-buf))
+        ;; Kill characters up to % in case we missed stripping prompts
+        ;; or +'s:
+        (goto-char (point-min))
+        (while (not (looking-at-p "%"))
```

Comment:
Why % is special? I think prompts are always on first line before new line and that is cleaned in ess-command.

Also a `(when (re-search-forward "%") (delete-region ...)))` logic would be cleaner and more efficient here I believe.

Reply:
The % is the first character in the roxygen output. Without this we miss a `+` so the preview buffer looks like:

```
+ % <output>
```

I'll change the looping logic later.